qatrfm/cli.py

```diff
@@ -87,6 +89,35 @@ def find_testcases(opt_test: Path):
     return load_testcases(basedir, files)
 
 
+def get_network_octet():
+    """
+    Find a non-used network in the system
+
+    To allow multiple environments co-exist, network ranges can't be
+    hardcoded. Otherwise, new libvirt virtual networks can't be created.
+    The default environment will create a network with range 10.X.0.0/24,
+    where X will be calculated dynamically according to the existing
+    networks on the system, starting from X=0, this offers 255 possible
+    isolated environments running at the same time.
+    """
+    locks_dir = '/tmp/qatrfm'
+    os.makedirs(locks_dir, exist_ok=True)
+    x = 0
+    while x < 255:
+        try:
+            f = open('{}/{}'.format(locks_dir, x), 'a')
+            [ret, output] = libutils.execute_bash_cmd('ip a|grep 10.{}||true')
+            if output == '':
+                fcntl.flock(f, fcntl.LOCK_EX | fcntl.LOCK_NB)
+                return f
```

Comment:
It will show you something like this (after a bunch of trials):

```
ll /tmp/qatrfm/
total 0
-rw-r--r-- 1 root root 0 Mar 15 17:09 0
-rw-r--r-- 1 root root 0 Mar 15 17:10 1
-rw-r--r-- 1 root root 0 Mar 15 17:12 10
-rw-r--r-- 1 root root 0 Mar 15 17:24 11
-rw-r--r-- 1 root root 0 Mar 15 17:24 12
-rw-r--r-- 1 root root 0 Mar 15 17:24 13
-rw-r--r-- 1 root root 0 Mar 15 17:24 14
-rw-r--r-- 1 root root 0 Mar 15 17:24 15
-rw-r--r-- 1 root root 0 Mar 15 17:24 16
-rw-r--r-- 1 root root 0 Mar 15 17:24 17
-rw-r--r-- 1 root root 0 Mar 15 17:24 18
-rw-r--r-- 1 root root 0 Mar 15 17:24 19
-rw-r--r-- 1 root root 0 Mar 15 17:12 2
-rw-r--r-- 1 root root 0 Mar 15 17:24 20
-rw-r--r-- 1 root root 0 Mar 15 16:07 3
-rw-r--r-- 1 root root 0 Mar 15 16:07 3.lock
-rw-r--r-- 1 root root 0 Mar 15 17:12 4
-rw-r--r-- 1 root root 0 Mar 15 17:12 5
-rw-r--r-- 1 root root 0 Mar 15 17:12 6
-rw-r--r-- 1 root root 0 Mar 15 17:12 7
-rw-r--r-- 1 root root 0 Mar 15 17:12 8
-rw-r--r-- 1 root root 0 Mar 15 17:12 9
```

Reply:
I think so too, that we can just keep the files. It's not nice, but I don't know a better way. Also if we do not have a global exeption/cleanup handler, there will be always the change of left overs.